### DISCUSSION TITLE
How to get installation requirements for Fedora atomic spins such as Silverblue etc

### DISCUSSION THREAD

**Discussion by quazar-omega** [quickemu-project/quickemu#1251] [Quickemu]
I'm trying to run this on Fedora Kinoite and I installed it using Nix Home Manager, but I can't start any virtual machines since I don't have all the necessary libvirt/kvm things installed.  

<details>

<summary>
For example, output for starting Windows 11 (getting a "core dumped" error):
</summary>

```
$ quickemu --vm windows-11.conf
Quickemu 4.9.4 using /nix/store/r8cdbvyh1ip36fr412lb80ca1cim5i08-qemu-8.2.4/bin/qemu-system-x86_64 v8.2.4
 - Host:     Fedora Linux 40.20240523.0 (Kinoite) running Linux 6.8 (fedora)
 - CPU:      AMD Ryzen 7 5700G with Radeon Graphics
 - CPU VM:   1 Socket(s), 4 Core(s), 2 Thread(s), 8G RAM
 - MSR:      WARNING! Ignoring unhandled Model-Specific Registers is disabled.

             echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs

             If you are unable to run macOS or Windows VMs then run the above 👆
             This will enable ignoring of unhandled MSRs until you reboot the host.
             You can make this change permanent by running: 'quickemu --ignore-msrs-always'
 - EFI:      WARNING! /usr/share/OVMF/OVMF_CODE.fd is a symlink.
             Resolving to... /usr/share/edk2/ovmf/OVMF_CODE.fd
 - BOOT:     EFI (Windows), OVMF (/usr/share/edk2/ovmf/OVMF_CODE.fd), SecureBoot (off).
 - Disk:     windows-11/disk.qcow2 (64G)
             Looks unused, booting from windows-11/Win11_23H2_English_x64v2.iso
 - Boot ISO: windows-11/Win11_23H2_English_x64v2.iso
 - CD-ROM:   windows-11/virtio-win.iso
 - Display:  SDL, virtio-vga-gl, GL (on), VirGL (on)
 - Sound:    intel-hda
 - ssh:      On host:  ssh user@localhost -p 22220
 - WebDAV:   On guest: dav://localhost:9843/
 - TPM:      windows-11/windows-11.swtpm-sock (6618)
 - Network:  User (virtio-net)
 - Monitor:  On host:  nc -U "windows-11/windows-11-monitor.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-monitor.socket
 - Serial:   On host:  nc -U "windows-11/windows-11-serial.socket"
             or     :  socat -,echo=0,icanon=0 unix-connect:windows-11/windows-11-serial.socket
 - Process:  Starting windows-11.conf as windows-11 (6636)
swtpm: Data client disconnected
/nix/store/zks11m3psf57lkjb67abks0s5cx96mmh-quickemu-4.9.4/bin/.quickemu-wrapped: line 2001:  6636 Aborted                 (core dumped) ${QEMU} "${args[@]}" &> "${VMDIR}/${VMNAME}.log"
 - Sending:  sendkey ret
```

</details>

I went through a [post on Fedora discussions](https://discussion.fedoraproject.org/t/overlaying-libvirt-on-silverblue-kinoite-sericea-onyx-and-coreos/86312/11) saying I could get away with installing just `qemu-kvm-core`, but I guess that's not enough after reading the [linked page](https://libvirt.org/kbase/rpm-deployment.html), what more should I try (if it's possible at all)?

Update 1: I've tried following another piece of advice on [this post](https://discussion.fedoraproject.org/t/is-possible-to-use-qemu-kvm-virtmanager-in-silverblue-without-layering/112511/8), so I tired to get virt-manager to run under Distrobox, that does work (almost fully), but I'm missing the piece that can link the libvirt in my container to quickemu in my host system (on Nix). Otherwise I guess I could install it within the container, but that's kinda bothersome and I'd like to avoid it if I can, any ideas?

Update 2: I thought that another alternative would be to layer the packages listed [here](https://discussion.fedoraproject.org/t/overlaying-libvirt-on-silverblue-kinoite-sericea-onyx-and-coreos/86312), and to still keep quickemu in Nix, but the error is the same, always on line `2001`, just a different error number (this time `11114`, it changes each time it's run though, so it doesn't matter I guess)

**Comment by quazar-omega** [comment]
Some actual progress, just randomly I tried to add the `--display spice` parameter to the command:

```sh
quickemu --vm windows-11.conf --display spice
```

And now the VM actually started without issues, but I don't understand why it did, and even if it is working, is this still advantageous or would the SDL display provide way better performance?
Also I'd need to narrow down what are the things I installed that were actually needed now.

**Comment by flexiondotorg** [comment]
SDL does provide slightly better GL acceleration in the guests, but SPICE is the much more versatile and useful option because USB pass-through, clipboard sharing, and file sharing are much easier to work with. 

In a future version of Quickemu, I will likely have it look for SPICE-capable viewers and default to those with a fallback to SDL.

**Reply by quazar-omega** [reply]
Oh, I see, then I guess I can keep using that for now!  
I am still curious about SDL though, do you know how I can get it working?
